internal/daemon/controller/handlers/authmethods/oidc.go

```diff
@@ -178,7 +178,14 @@ func (s Service) authenticateOidcStart(ctx context.Context, req *pbs.Authenticat
 	}
 
 	authUrl, tokenId, err := oidc.StartAuth(ctx, s.oidcRepoFn, req.GetAuthMethodId(), opts...)
-	if err != nil {
+	switch {
+	case errors.Match(errors.T(errors.AuthMethodInactive), err):
+		return nil, handlers.ApiErrorWithCodeAndMessage(codes.FailedPrecondition, "Cannot start authentication against an inactive OIDC auth method")
```

Comment:
A note that this actually returns a HTTP 400 error from the [GRPC gateway](https://github.com/grpc-ecosystem/grpc-gateway/blob/93189d015a3e23872723445862dc6fe1c4502913/runtime/errors.go#L58-L60). I assume this would still make sense since it is on the client to activate the OIDC auth method, does that sound correct?

Reply:
Alternatively, could the GRPC `Aborted` error could be more suitable here? It maps to the `HTTP.StatusConflict`,` which is more descriptive to the case but not sure `Aborted` is more suitable than `FailedPrecondition` here

https://github.com/grpc-ecosystem/grpc-gateway/blob/93189d015a3e23872723445862dc6fe1c4502913/runtime/errors.go#L61-L62

Reply:
I just saw that @johanbrandhorst was having a relevant discussion. This might be useful context, not sure if 400 is suitable here

Reply:
Oh that's interesting, I wasn't aware that was the mapping behaviour. I'm personally fine with a regular 400.

Reply:
Yes, the mapping is 400, I think that's fine here.

Reply:
I've updated the commit & PR title/description to reflect this is a 400 not a 412.